contracts/EVMScriptFactories/AddNodeOperators.sol

```diff
@@ -0,0 +1,195 @@
+
+    // -------------
+    // CONSTRUCTOR
+    // -------------
+
+    constructor(
+        address _trustedCaller,
+        address _nodeOperatorsRegistry,
+        address _acl
+    ) TrustedCaller(_trustedCaller) {
+        nodeOperatorsRegistry = INodeOperatorsRegistry(_nodeOperatorsRegistry);
+        acl = IACL(_acl);
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    function createEVMScript(
+        address _creator,
+        bytes memory _evmScriptCallData
+    ) external view override onlyTrustedCaller(_creator) returns (bytes memory) {
+        (
+            uint256 nodeOperatorsCount,
+            AddNodeOperatorInput[] memory decodedCallData
+        ) = _decodeEVMScriptCallData(_evmScriptCallData);
+
+        address[] memory toAddresses = new address[](decodedCallData.length * 2);
+        bytes4[] memory methodIds = new bytes4[](decodedCallData.length * 2);
+        bytes[] memory encodedCalldata = new bytes[](decodedCallData.length * 2);
+
+        _validateInputData(nodeOperatorsCount, decodedCallData);
+
+        for (uint256 i = 0; i < decodedCallData.length; i++) {
+            toAddresses[i * 2] = address(nodeOperatorsRegistry);
+            methodIds[i * 2] = ADD_NODE_OPERATOR_SELECTOR;
+            encodedCalldata[i * 2] = abi.encode(
+                decodedCallData[i].name,
+                decodedCallData[i].rewardAddress
+            );
+
+            // See https://legacy-docs.aragon.org/developers/tools/aragonos/reference-aragonos-3#parameter-interpretation for details
+            uint256[] memory permissionParams = new uint256[](1);
+            permissionParams[0] = (1 << 240) + nodeOperatorsCount + i;
+
+            toAddresses[i * 2 + 1] = address(acl);
+            methodIds[i * 2 + 1] = GRANT_PERMISSION_P_SELECTOR;
+            encodedCalldata[i * 2 + 1] = abi.encode(
+                decodedCallData[i].managerAddress,
+                address(nodeOperatorsRegistry),
+                MANAGE_SIGNING_KEYS_ROLE,
+                permissionParams
+            );
+        }
+
+        return EVMScriptCreator.createEVMScript(toAddresses, methodIds, encodedCalldata);
+    }
+
+    function decodeEVMScriptCallData(
+        bytes memory _evmScriptCallData
+    )
+        external
+        pure
+        returns (uint256 nodeOperatorsCount, AddNodeOperatorInput[] memory nodeOperators)
+    {
+        return _decodeEVMScriptCallData(_evmScriptCallData);
+    }
+
+    // ------------------
+    // PRIVATE METHODS
+    // ------------------
+
+    function _decodeEVMScriptCallData(
+        bytes memory _evmScriptCallData
+    )
+        private
+        pure
+        returns (uint256 nodeOperatorsCount, AddNodeOperatorInput[] memory nodeOperators)
+    {
+        (nodeOperatorsCount, nodeOperators) = abi.decode(
+            _evmScriptCallData,
+            (uint256, AddNodeOperatorInput[])
+        );
+    }
+
+    function _validateInputData(
```

Comment:
Consider adding a check of existing running motion that have the same factory address (iterate through the motions and check `evmScriptFactory`)